ethcore/src/client/test_client.rs

```diff
@@ -611,8 +611,8 @@ impl BlockChainClient for TestBlockChainClient {
 		self.execution_result.read().clone().unwrap()
 	}
 
-	fn replay_block_transactions(&self, _block: BlockId, _analytics: CallAnalytics) -> Result<Box<Iterator<Item = Executed>>, CallError> {
-		Ok(Box::new(self.execution_result.read().clone().unwrap().into_iter()))
+	fn replay_block_transactions(&self, _block: BlockId, _analytics: CallAnalytics) -> Result<Box<Iterator<Item = (H256, Executed)>>, CallError> {
+		Ok(Box::new(self.traces.read().clone().unwrap().into_iter().map(|t| t.transaction_hash.unwrap()).zip(self.execution_result.read().clone().unwrap().into_iter())))
```

Comment:
@shamardy Sorry if I wasn't clear -- I think this should be changed **not** to use `self.traces` (but just return `H256::new`), because when we query `self.execution_result` in test, `self.traces` is never available -- `self.execution_result` is only set by `self.set_execution_result` and we only have a few appearance our code.

Reply:
The above code will cause testing run of replayBlockTransactions to always return empty vector -- definitely not something we want.